js/vnext/projection/selection.js

```diff
@@ -65,16 +65,17 @@ export function selection (state, selection) {
       single: model.single,
       checked: this.countRows > 0 && model.selected.length === this.countRows,
     }),
+    template: (index) => selectionBodyTemplate({
+          single: model.single,
+          checked: selectedIndex[index],
+        }),
     sortable: false,
   }].concat(state.columns);
 
   const bodyRows = state.bodyRows.map((row, index) => _.defaults({
     item: _.defaults({
       selection: {
-        html: selectionBodyTemplate({
-          single: model.single,
-          checked: selectedIndex[index],
-        }),
+        index,
```

Comment:
We should avoid adding things to `item` for now, since the `item` is supposed to be the same as the original model. It's possible for the original model having a `selection` property.

We should pass the index in from the caller.